ERCS/erc-7580.md

```diff
@@ -0,0 +1,92 @@
+
+## Specification
+
+The keywords “MUST,” “MUST NOT,” “REQUIRED,” “SHALL,” “SHALL NOT,” “SHOULD,” “SHOULD NOT,” “RECOMMENDED,” “MAY,” and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+**Interface**
+
+This protocol standardizes how to keep tracks on inter dapps operations, which initially offers 2 main methos onTrackStart and onTrackEnd.
+
+```solidity
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.10;
+
+interface IERCXXX {
+    // Events
+
+    /// @dev Emits when track starts.
+    /// @param track_id track id
+    /// @param contract_address the address of tracking contract
+    /// @param function_hash the hash of tracking function
+    event onTrackStartRecorded(uint256 track_id, address contract_address, bytes32 function_hash);
+
+    /// @dev Emits when track ends.
+    /// @param owner The owner address of the newly minted combo
+    /// @param comboId The newly minted combo identifier
+    event onTrackStartRecorded(uint256 track_id, address contract_address, bytes32 function_hash);
+
+    // Functions
+
+    /// @dev Track a specified contract function start move.
+    /// @param track_id track id
+    /// @param contract_address the address of tracking contract
+    /// @param function_hash the hash of tracking function
+    function onTrackStart(uint256 track_id, address contract_address, bytes32 function_hash) external;
+
+    /// @dev Track a specified contract function end move.
+    /// @param track_id track id
+    /// @param contract_address the address of tracking contract
+    /// @param function_hash the hash of tracking function
+    function onTrackEnd(uint256 track_id, address contract_address, bytes32 function_hash);
+
+    // TODO: may need more
+}
+```
+
+**Routine**
+
+Imaging we have two roles here: promotion/ads business role and projects role who adopted the promotion service. And project contracts implemented the above interfaces. Then we would have a routine:
+1. projects get a seed id (hash) from promotion side
+2. before the target promotion action starts, project contracts called the interface ‘onTrackStart(id, contract_address, function_hash)’
+3. after the target promotion action ends, project contracts called the inferface ‘onTrackEnd(id, contract_address, function_hash)’
+4. promotion contract collect the project action info and distribute the rewards back to projects
+
+
+## Rationale
+
+The core mechanism for this proposal is to provide a shared track interface standard for inter dapps, to improve the efficiency and fulfile the required tracking business. We provide two neaty interface functions onTrackStart and onTrackEnd to fill the basic required info and connect the necessary dots. Sure there're more demands for more functions and it would be updated later.
```

Comment:
This reads more like Motivation than Rationale. The Motivation section justifies the proposal as a whole, while the Rationale should explain technical choices made within the proposal.

In other words:

> Motivation: "Why do we need a shed?"
> Rationale: "Why is the shed red?"